overview.php

```diff
@@ -169,6 +171,22 @@
 }
 echo '&nbsp;' . get_string('role') . '&nbsp;';
 echo $OUTPUT->single_select($PAGE->url, 'role', $rolestodisplay, $roleselected);
+
+// Output the sections menu.
+$sql = "SELECT DISTINCT s.id, s.section
+          FROM {course_sections} s
+         WHERE s.course = :courseid
+           AND s.sequence <> ''";
+$params = array('courseid' => $courseid);
+$sectionrecords = $DB->get_records_sql($sql, $params);
+$sections = array_values($sectionrecords);
+$numberofsections = count($sections);
+$sectionstodisplay = array(0 => get_string('allsections', 'block_completion_progress'));
+for ($i = 0; $i < $numberofsections; $i++) {
+    $sectionstodisplay[$sections[$i]->id] = get_section_name($courseid, $sections[$i]->section);
+}
+echo '&nbsp;' .get_string('section') . '&nbsp;';
+echo $OUTPUT->single_select($PAGE->url, 'section', $sectionstodisplay, $section);
```

Comment:
I suggest using this instead:

    // Output the sections menu.
    $format = course_get_format($COURSE);
    if ($format->uses_sections()) {
        $sections = [ -1 => get_string('allsections', 'block_completion_progress') ];
        foreach ($format->get_sections() as $sectinfo) {
            $sections[$sectinfo->section] = $format->get_section_name($sectinfo);
        }
        echo '&nbsp;' . get_string('section') . '&nbsp;';
        echo $OUTPUT->single_select($PAGE->url, 'section', $sections, $section);
    }

Reply:
Other adjustments as a consequence of using the section number instead of its record ID are obviously necessary with that change.

Reply:
Thanks I will check that :)